atomic_reactor/tasks/common.py

```diff
@@ -127,7 +121,14 @@ def run(self, *args, **kwargs):
             else:
                 signal.signal(signal.SIGTERM, self.throw_task_canceled_exception)
 
-            self.execute(*args, **kwargs)
+            result = self.execute(*args, **kwargs)
+            if self._params.task_result:
+                write_task_result(self._params.task_result, repr(result))
+
+        except Exception as e:
+            if self._params.task_result:
+                write_task_result(self._params.task_result, repr(e))
+            raise(e)
```

Comment:
Do we have unit tests for writing the results? I couldn't find one

Reply:
added